modules/configuration-annotation-processor/src/main/java/org/apache/ignite/internal/configuration/processor/validation/Validator.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.configuration.processor.validation;
+
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toSet;
+import static org.apache.ignite.internal.configuration.processor.ConfigurationProcessor.TOP_LEVEL_ANNOTATIONS;
+import static org.apache.ignite.internal.configuration.processor.ConfigurationProcessorUtils.containsAnyAnnotation;
+import static org.apache.ignite.internal.configuration.processor.ConfigurationProcessorUtils.isValidValueAnnotationFieldType;
+import static org.apache.ignite.internal.configuration.processor.ConfigurationProcessorUtils.joinSimpleName;
+import static org.apache.ignite.internal.configuration.processor.ConfigurationProcessorUtils.sameType;
+import static org.apache.ignite.internal.configuration.processor.ConfigurationProcessorUtils.simpleName;
+
+import java.lang.annotation.Annotation;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import javax.annotation.processing.ProcessingEnvironment;
+import javax.lang.model.element.Name;
+import javax.lang.model.element.VariableElement;
+import org.apache.ignite.internal.configuration.processor.ClassWrapper;
+import org.apache.ignite.internal.configuration.processor.ConfigurationValidationException;
+
+/**
+ * Base class for configuration validators.
+ */
+public abstract class Validator {
```

Comment:
Maybe it would be better to make an interface out of it?

Reply:
We will not be able to store ProcessingEnvironment as a field then, do you propose to move it to subclasses?

Reply:
Why not, or maybe rename to something like "AbstractConfigurationClassValidator".

Reply:
How is this better than the current approach?

Reply:
Thank you, I will leave the current approach then